src/SamplesApp/SamplesApp.UITests/TestFramework/ImageAssert.cs

```diff
@@ -15,159 +15,110 @@ namespace SamplesApp.UITests.TestFramework
 
-		private static void AreEqualImpl(FileInfo expected, Rectangle expectedRect, FileInfo actual, Rectangle actualRect, int line)
-		{
-			Assert.AreEqual(expectedRect.Size, actualRect.Size, WithContext("Compare rects don't have the same size"));
-
-			using (var expectedBitmap = new Bitmap(expected.FullName))
-			using (var actualBitmap = new Bitmap(actual.FullName))
-			{
-				Assert.AreEqual(expectedBitmap.Size, actualBitmap.Size, WithContext("Screenshots don't have the same size"));
-
-				var width = Math.Min(expectedRect.Width, expectedBitmap.Size.Width);
-				var height = Math.Min(expectedRect.Height, expectedBitmap.Size.Height);
-
-				(int x, int y) expectedOffset = (
-					expectedRect.X < 0 ? expectedBitmap.Size.Width + expectedRect.X : expectedRect.X,
-					expectedRect.Y < 0 ? expectedBitmap.Size.Height + expectedRect.Y : expectedRect.Y
-				);
-				(int x, int y) actualOffset = (
-					actualRect.X < 0 ? actualBitmap.Size.Width + actualRect.X : actualRect.X,
-					actualRect.Y < 0 ? actualBitmap.Size.Height + actualRect.Y : actualRect.Y
-				);
-
-				for (var x = 0; x < width; x++)
-					for (var y = 0; y < height; y++)
-					{
-						var expectedColor = expectedBitmap.GetPixel(x + expectedOffset.x, y + expectedOffset.y);
-						var actualColor = actualBitmap.GetPixel(x + actualOffset.x, y + actualOffset.y);
-
-						if (expectedColor != actualColor)
-						{
-							Assert.Fail(WithContext(builder: builder => builder
-								.AppendLine($"Pixels at ({x}, {y}) are not the same")
-								.AppendLine($"expectedColor: {ToArgbCode(expectedColor)} {expectedColor}")
-								.AppendLine($"actualColor  : {ToArgbCode(actualColor)} {actualColor}")
-							));
-						}
-					}
-			}
-
-			string WithContext(string message = null, Action<StringBuilder> builder = null)
-			{
-				return new StringBuilder()
-					.AppendLine($"ImageAssert.AreEqual @ line {line}")
-					.AppendLine("expected: " + expected?.Name + (expectedRect == FirstQuadrant ? null : $"in {expectedRect}"))
-					.AppendLine("actual  : " + actual?.Name + (actualRect == FirstQuadrant ? null : $"in {actualRect}"))
-					.AppendLine("====================")
-					.ApplyIf(message != null, x => x.AppendLine(message))
-					.ApplyIf(builder != null, builder)
-					.ToString();
-			}
-		}
-		#endregion
-
-		#region AreAlmostEqual
 		/// <summary>
 		/// Asserts that two screenshots are equal to each other inside the area of the nominated rect to within the given pixel error
 		/// (ie, the A, R, G, or B values cumulatively differ by more than the permitted error for any pixel).
 		/// </summary>
 		public static void AreAlmostEqual(FileInfo expected, FileInfo actual, IAppRect rect, int permittedPixelError, [CallerLineNumber] int line = 0)
-			=> AreAlmostEqualImpl(expected, rect.ToRectangle(), actual, rect.ToRectangle(), permittedPixelError, line);
+			=> AreEqualImpl(expected, rect.ToRectangle(), actual, rect.ToRectangle(), 1, PixelTolerance.Cummulative(permittedPixelError), line);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'rect' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5363624)